ENGAGEHF/Managers/NotificationManager/NotificationManager.swift

```diff
@@ -20,41 +22,88 @@ import UserNotifications
 @MainActor
 class NotificationManager: Module, NotificationHandler, NotificationTokenHandler, EnvironmentAccessible {
     @ObservationIgnored @Application(\.registerRemoteNotifications) private var registerRemoteNotifications
+    @ObservationIgnored @Application(\.unregisterRemoteNotifications) private var unregisterRemoteNotifications
+    @ObservationIgnored @Dependency(AccountNotifications.self) private var accountNotifications: AccountNotifications?
     @ObservationIgnored @Application(\.logger) private var logger
     @ObservationIgnored @Dependency(NavigationManager.self) private var navigationManager
     
     @ObservationIgnored @AppStorage(StorageKeys.onboardingFlowComplete) private var completedOnboardingFlow = false
+    @ObservationIgnored @Environment(Account.self) private var account: Account?
     
     
     private var cancellable: AnyCancellable?
+    private var notificationsTask: Task<Void, Never>?
+    
+    private var apnsDeviceToken: Data?
```

Comment:
As far as I can see this would only be set at the first time when you register for notifications, subsequent launches of the app don't get this information as we never pull it. We might check out if we either store it in the app storage or secure storage? Ideally we don't have to store it at all if the register device token function returns a stable token every time we call it.